src/main-client.js

```diff
@@ -117,4 +117,95 @@ function getPastBookingsOnDom(bookings, rooms, todaysDate) {
+              travel time. Please accept our sincerest apologies for the inconvenience.</p>
+          </section>`)
+    } else {
+      $('.fierce-apology-container').css('display', 'none')
+      $('.room-booking-container').css('display', 'flex')
+      availableRooms.forEach(room => {
+        $('.rooms-available-display').append(`
+        <div class="single-room-info" id="${room.number}">
+        <div class="single-room-heading-wrapper">
+        <h6 class="single-room-heading">${room.roomType}</h6>
+        </div>
+        <section class="single-room-description">
+        <p class="single-room-text">Our ${room.roomType}  provides views over landscaped gardens. Room number ${room.number} has a
+        seating area, ample storage, digital safe, minibar, and luxurious duck down bedding.This room
+        comes with ${room.numBeds} ${room.bedSize} bed and may or may not have a bidet. The entire suite is designed in a fashion to
+        allow mobility impaired accessibility. Price per night: ${room.costPerNight}</p>
+        <button class="book-room-button">Select Room</button>
+        </section>
+        </div>
+        `)
+      })
+      $('.submit-option-button').on('click', {availableRooms: availableRooms, date: date}, filterRoomByType)
+    }
+  }
+  
+  function filterRoomByType(jqueryEvent) {
+    let rooms = jqueryEvent.data.availableRooms;
+    let date = jqueryEvent.data.date;
+    let selectedOption = $('.select-container').val()
+    $('.rooms-available-display').text('')
+    rooms.forEach(room => {
+      if(room.roomType === selectedOption) {
+        $('.rooms-available-display').append(`
+        <div class="single-room-info" id="${room.number}">
+        <div class="single-room-heading-wrapper">
+        <h6 class="single-room-heading">${room.roomType}</h6>
+        </div>
+        <section class="single-room-description">
+        <p class="single-room-text">Our ${room.roomType}  provides views over landscaped gardens. Room number ${room.number} has a
+        seating area, ample storage, digital safe, minibar, and luxurious duck down bedding.This room
+        comes with ${room.numBeds} ${room.bedSize} bed and may or may not have a bidet. The entire suite is designed in a fashion to
+        allow mobility impaired accessibility. Price per night: ${room.costPerNight}</p>
+        <button class="book-room-button">Select Room</button>
+        </section>
+        </div>
+        `)
+      }
+    })
+    urlString = window.location.href
+    url = new URL(urlString)
+    userId = url.searchParams.get('userId')
+    $('.book-room-button').on('click', {date: date, userId: userId}, selectRoom)
+  }
+
+  function selectRoom(jqueryEvent) {
+    fetch('https://fe-apps.herokuapp.com/api/v1/overlook/1904/bookings/bookings', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        "userID": Number(jqueryEvent.data.userId),
```

Comment:
Interesting use of jQuery for the capture, cool!

Reply:
Yeah I was stumped on event listeners all day yesterday and didn't know how to pass that data through on the function I was calling so I did some research and found that jquery has it's own way of doing it.